dapp_methods_calls.py

```diff
@@ -0,0 +1,49 @@
+from pyteal import *
+
+
+def methods_calls(methods: dict):
+
+    args = []
+    for method_name, method_call in methods.items():
+        args += [[Txn.application_args[0] == Bytes(method_name), method_call]]
+
+    precondition = And(
+        Txn.type_enum() == TxnType.ApplicationCall,
+        Txn.rekey_to() == Global.zero_address(),
+        Txn.application_args.length() >= Int(1),
+    )
+
+    return Seq([
+        Assert(precondition),
+        Cond(*args),
+        Return(Int(1)),
```

Comment:
if none of the calls match should this fail?

Reply:
ah 
> If no condition evalutes to a true value, the Cond expression produces an error and the TEAL program terminates.

This should be fine then

Reply:
Yes, and this is ensured by `Cond(*args)`.